console/v1/types_console_sample.go

```diff
@@ -0,0 +1,261 @@
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=50
+	Title string `json:"title"`
+
+	// abstract is a short introduction to the sample.
+	//
+	// It is required and must be no more than 100 characters in length.
+	//
+	// The abstract is shown on the sample card tile below the title and provider
+	// and is limited to three lines of content.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MaxLength=100
+	Abstract string `json:"abstract"`
+
+	// description is a long form explanation of the sample.
+	//
+	// It is required and can have a maximum length of **4096** characters.
+	//
+	// It is a README.md-like content for additional information, links, pre-conditions, and other instructions.
+	// It will be rendered as Markdown so that it can contain line breaks, links, and other simple formatting.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MaxLength=4096
+	Description string `json:"description"`
+
+	// icon is an optional base64 encoded image and shown beside the sample title.
+	//
+	// The format must follow the [data: URL format] and can have a maximum size of **10 KB**.
+	//
+	//   data:[<mediatype>][;base64],<base64 encoded image>
+	//
+	// For example:
+	//
+	//   data:image;base64,             plus the base64 encoded image.
+	//
+	// Vector images can also be used. SVG icons must start with:
+	//
+	//   data:image/svg+xml;base64,     plus the base64 encoded SVG image.
+	//
+	// All sample catalog icons will be shown on a white background (also when the dark theme is used).
+	// The web console ensures that different aspect radios work correctly.
+	// Currently, the surface of the icon is at most 40x100px.
+	//
+	// [data: URL format]: https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/Data_URLs
+	// +optional
+	// +kubebuilder:validation:Pattern=`^data:([a-z/\.+0-9]*;(([-a-zA-Z0-9=])*;)?)?base64,`
+	// +kubebuilder:validation:MaxLength=14000
+	Icon string `json:"icon"`
+
+	// type is an optional label to group multiple samples.
+	//
+	// It is optional and must be no more than 20 characters in length.
+	//
+	// Recommendation is a singular term like "Builder Image", "Devfile" or "Serverless Function".
+	//
+	// Currently, the type is shown a badge on the sample card tile in the top right corner.
+	// +optional
+	// +kubebuilder:validation:MaxLength=20
+	Type string `json:"type"`
+
+	// provider is an optional label to honor who provides the sample.
+	//
+	// It is optional and must be no more than 50 characters in length.
+	//
+	// A provider can be a company like "Red Hat" or an organization like "CNCF" or "Knative".
+	//
+	// Currently, the provider is only shown on the sample card tile below the title with the prefix "Provided by "
+	// +optional
+	// +kubebuilder:validation:MaxLength=50
+	Provider string `json:"provider"`
+
+	// tags like "Java", "Quarkus", etc.
+	// Tags are optional and can be used by the user to find samples in the samples catalog.
```

Comment:
This should be reversed, introduce what a tag is, then give examples
```suggestion
	// tags are optional string values that can be used to find samples in the samples catalog.
	// Examples of common tags may be "Java", "Quarkus", etc.
```